src/nav/navActions.js

```diff
@@ -103,3 +103,14 @@ export const navigateToLegal = (): NavigationAction => StackActions.push({ route
 
 export const navigateToUserStatus = (): NavigationAction =>
   StackActions.push({ routeName: 'user-status' });
+
+export const navigateToUserScreenForUserId = (userId: number) => (
+  dispatch: Dispatch,
+  getState: GetState,
+) => {
+  const state = getState();
+  const user = getUserForId(state, userId);
+  if (user && user.email) {
+    dispatch(navigateToAccountDetails(user.email));
+  }
```

Comment:
If this fails, it shouldn't do so silently; there should be some sort of notification (`Alert.alert`, perhaps?) and error handling (at least `logging.warn`).

Note that a failure here does not necessarily indicate a bug! I believe it may be possible, on some servers, to craft an @&zwj;-mention containing an invalid or unused user-id. (Or at least have historically been possible, in which case there may still be messages floating around in the history with invalid user-ids.)